kobo/hub/views.py

```diff
@@ -247,15 +244,19 @@ class LogoutView(django.contrib.auth.views.LogoutView):
 
 
 def krb5login(request, redirect_field_name=REDIRECT_FIELD_NAME):
-    #middleware = 'django.contrib.auth.middleware.RemoteUserMiddleware'
     middleware = 'kobo.django.auth.middleware.LimitedRemoteUserMiddleware'
     if middleware not in settings.MIDDLEWARE:
         raise ImproperlyConfigured("krb5login view requires '%s' middleware installed" % middleware)
-    redirect_to = request.POST.get(redirect_field_name, "")
-    if not redirect_to:
-        redirect_to = request.GET.get(redirect_field_name, "")
-    if not redirect_to:
-        redirect_to = reverse("home/index")
+
+    redirect_to = request.POST.get(redirect_field_name, request.GET.get(redirect_field_name))
+    url_is_safe = url_has_allowed_host_and_scheme(
+            url=redirect_to,
+            allowed_hosts=request.get_host(),
+            require_https=request.is_secure(),
+    )
+    if not url_is_safe:
+        redirect_to = reverse(settings.LOGIN_REDIRECT_URL)
```

Comment:
Updating the settings.py.template doesn't update the actual settings.py deployed by anyone using kobo, so this setting is going to be missing in almost all cases, right?  Should it apply a default rather than crashing?

Reply:
This line of code should not crash.  According to Django docs, unset settings are set to the default value (but I admit that the default path is non-existent in `kobo` based projects): https://github.com/django/django/blob/07f9f9960c477c7f60a5f3f243a0f404a56b1121/django/conf/global_settings.py#L532C9-L532C9 

Also, some users may already have `LOGIN_REDIRECT_URL` set to their custom value, because the `LOGIN_REDIRECT_URL` setting was already used by the password login view when accessed with an unsafe redirect.